c/examples/tree_iteration.c

```diff
@@ -25,20 +25,20 @@ main(int argc, char **argv)
     check_tsk_error(ret);
 
     printf("Iterate forwards\n");
-    for (iter = tsk_tree_first(&tree); iter == 1; iter = tsk_tree_next(&tree)) {
+    for (ret = tsk_tree_first(&tree); ret == TSK_TREE_OK; ret = tsk_tree_next(&tree)) {
```

Comment:
I like this change, but there's a few places it is missing:
```
test_stats.c:634
test_trees.c:303,321,352,367,6481
halotype_matching.c:982,1345
stats.c:195,232
trees.c:5448,5468
```
I also wonder if `tsk_diff_iter_next` should use the same convention?